linera-views/src/memory.rs

```diff
@@ -62,17 +73,17 @@ impl ReadableKeyValueStore<MemoryStoreError> for MemoryStore {
     }
 
     async fn read_value_bytes(&self, key: &[u8]) -> Result<Option<Vec<u8>>, MemoryStoreError> {
-        let map = self.map.read().await;
+        let map = self.map.read()?;
```

Comment:
`.unwrap()` is fine, or `.expect("lock should not be poisoned")`

Typically, poisoned locks are a valid reason to panic (because another thread has already panicked).

Reply:
Ok, I thought it was fine and better to eliminate the `.unwrap()`.

Reply:
This `.lock().unwrap()` is one of the only exceptions